bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/LinkImpl.java

```diff
@@ -143,20 +142,64 @@ public String getAssetPath() {
     }
 
     private Map<String, String> getQueryParams() {
-        if (queryParamsMap == null) {
+        if (queryParamsMap == null || this.queryParamsMap.isEmpty()) {
             populateQueryParams();
         }
         return queryParamsMap;
     }
 
     private void populateQueryParams() {
+        this.queryParamsMap = new HashMap<String, String>();
```

Comment:
Are we filtering the query params against XSS attacks, since this could cause a server side vulnerability ?

Reply:
Nope. This is being handled by sightly at https://github.com/virresh/aem-core-forms-components/blob/link_functionalityedit/ui.apps/src/main/content/jcr_root/apps/core/fd/components/formsportal/link/v1/link/link.html#L24 .
I just tested with XSS parsing at server side and it leads to double escaping which changes the meaning of query parameter
Should we instead handle XSS filtering here and pass unsafe context on the sightly or let sightly handle it?

Reply:
Letting sightly handle XSS filtering for now